staging/vhost-device-gpu/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Red Hat Inc
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+#![deny(
+    clippy::undocumented_unsafe_blocks,
+    /* groups */
+    clippy::correctness,
+    clippy::suspicious,
+    clippy::complexity,
+    clippy::perf,
+    clippy::style,
+    clippy::nursery,
+    //* restriction */
+    clippy::dbg_macro,
+    clippy::rc_buffer,
+    clippy::as_underscore,
+    clippy::assertions_on_result_states,
+    //* pedantic */
+    clippy::cast_lossless,
+    clippy::cast_possible_wrap,
+    clippy::ptr_as_ptr,
+    clippy::bool_to_int_with_if,
+    clippy::borrow_as_ptr,
+    clippy::case_sensitive_file_extension_comparisons,
+    clippy::cast_lossless,
+    clippy::cast_ptr_alignment,
+    clippy::naive_bytecount
+)]
+#![allow(
+    clippy::significant_drop_in_scrutinee,
+    clippy::significant_drop_tightening
+)]
+
+#[cfg(target_env = "gnu")]
+pub mod device;
+#[cfg(target_env = "gnu")]
+pub mod protocol;
+#[cfg(target_env = "gnu")]
+pub mod virtio_gpu;
+
+use std::path::PathBuf;
+
+use clap::ValueEnum;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, ValueEnum)]
+pub enum GpuMode {
+    VirglRenderer,
+    Gfxstream,
+}
+
+#[derive(Debug, Clone)]
+/// This structure is the public API through which an external program
```

Comment:
I am still curious what is really the difference between this struct and GpuArgs. It seems to be the exact same thing, so any reason for duplicating it?

Reply:
Well, sure. But as those crates are binary crates, I don't understand where the external configuration can come from? Unless a CLI configuration can be different from the in-code one, I don't see what is the need behind such duplication. 

I will let vhost-device maintainers to decide as this is just a drive-by comment.

Reply:
We want the crates to be able to used as libraries, because there's no reason not to. I think what dorinda and matej did is what we should be doing in general

Reply:
Unfortunately, you can't have a dependency (clap) only for the binary side, so my question about duplicating the struct still stands as both the library and the binary share the clap dependency at the end.

Reply:
IMHO, I prefer to keep it, they're many benefits for example it'll allow us have the flexibility to adapt it with additional fields, and defaults that are not tied to the command-line arguments. in the future we plan to have more field that could also have a default.

Reply:
Marking as resolved, if you are not satisfied feel free to reopen it.

Reply:
Yep sure, i am not fully aware of vhost-device usages so my comments are generic, they are not from a maintainer perspective obviously;)